extensions/wikia/FilePage/WikiaFilePage.php

```diff
@@ -153,6 +153,8 @@ protected function newPage( Title $title ) {
 	 */
 	public function view() {
 		global $wgRedirectFilePagesForAnons;
+		Wikia::setSurrogateKeysHeaders( FilePageHelper::getSurrogateKeys( $this->getTitle()	) ,
```

Comment:
if we decide to include file pages redirects only on selected wikis, this should be replaces with a hook call.